### PR TITLE
Add run mode to kvutils to dump index metadata and exit

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  when(MockedLedgerFactory.manipulateConfig(any[Config[Unit]])).thenAnswer[Config[Unit]](identity)
+
+  private val DumpIndexMetadataCommand = "dump-index-metadata"
+
+  behavior of "Runner"
+
+  it should "fail if a participant is not provided in run mode" in {
+
+    val runner = new Runner("Test", MockedLedgerFactory).owner(Seq.empty)
+    runner.use(_ => Future.unit).map(_ => fail).recover { case NonFatal(_) => succeed }
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala:31]: Perhaps use `.failed` here and below?

```suggestion
    runner.use(_ => Future.unit).failed.map(_ => succeed)
```

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  when(MockedLedgerFactory.defaultExtraConfig).thenReturn(())
+  when(MockedLedgerFactory.manipulateConfig(any[Config[Unit]])).thenAnswer[Config[Unit]](identity)
+
+  private val DumpIndexMetadataCommand = "dump-index-metadata"
+
+  behavior of "Runner"
+
+  it should "fail if a participant is not provided in run mode" in {
+
+    val runner = new Runner("Test", MockedLedgerFactory).owner(Seq.empty)
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala:30]: This feels like a very high level to test this behavior. Why not add it to the `CliSpec`?

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/7213/commits/6e8283ff951a96f6b60752a9b86dbc47ec948e39

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/7213/commits/b157bb878277d49a3f0921b6958f5867d9395897

**Review by SamirTalwar-DA — APPROVED** [review]
Looks great.

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/ConfigSpec.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.scalatest.{FlatSpec, Matchers, OptionValues}
+import scopt.OptionParser
+
+final class ConfigSpec extends FlatSpec with Matchers with OptionValues {
+
+  private val DumpIndexMetadataCommand = "dump-index-metadata"
+  private val ConfigParser: Seq[String] => Option[Config[Unit]] =
+    Config.parse("Test", (_: OptionParser[Config[Unit]]) => (), (), _)
+
+  behavior of "Runner"
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/ConfigSpec.scala:15]: Shouldn't this be `ConfigParser`?

**Pull request merged**